src/protocol/decoder.js

```diff
@@ -56,10 +56,34 @@ module.exports = class Decoder {
     return this.canReadBytes(INT64_SIZE)
   }
 
+  // Implementation taken from Node 12
+  // https://github.com/nodejs/node/blob/923d8bc733262cf960b62a02f113cfb0412b5834/lib/internal/buffer.js#L140
+  readBigInt64BE(buffer, offset = 0) {
+    if (isNaN(offset)) return BigInt(0)
+    const first = buffer[offset]
+
+    const last = buffer[offset + 7]
+
+    if (first === undefined || last === undefined) {
+      throw Error(offset, buffer.length - 8)
+    }
+
+    const val =
+      (first << 24) + // Overflow
```

Comment:
Hmm, what does the comment here mean?

Reply:
Also, either use bit shifting or multiplication + exponentiation everywhere, instead of using bit shifting for the first line and multi+exponent for the remaining lines. Took me a second to understand that `n << 24 === n * 2 ** 24`.